multiversx_sdk/core/controllers/smart_contract_controller.py

```diff
@@ -0,0 +1,226 @@
+from pathlib import Path
+from typing import Any, List, Optional, Protocol, Sequence, Union
+
+from multiversx_sdk.adapters.query_runner_adapter import QueryRunnerAdapter
+from multiversx_sdk.converters.transactions_converter import \
+    TransactionsConverter
+from multiversx_sdk.core.controllers.network_provider_wrapper import \
+    ProviderWrapper
+from multiversx_sdk.core.interfaces import IAddress
+from multiversx_sdk.core.smart_contract_queries_controller import \
+    SmartContractQueriesController
+from multiversx_sdk.core.tokens import TokenTransfer
+from multiversx_sdk.core.transaction import Transaction
+from multiversx_sdk.core.transaction_computer import TransactionComputer
+from multiversx_sdk.core.transactions_factories.smart_contract_transactions_factory import \
+    SmartContractTransactionsFactory
+from multiversx_sdk.core.transactions_factories.transactions_factory_config import \
+    TransactionsFactoryConfig
+from multiversx_sdk.core.transactions_outcome_parsers.smart_contract_transactions_outcome_parser import \
+    SmartContractTransactionsOutcomeParser
+from multiversx_sdk.core.transactions_outcome_parsers.smart_contract_transactions_outcome_parser_types import \
+    SmartContractDeployOutcome
+from multiversx_sdk.network_providers.transaction_awaiter import \
+    TransactionAwaiter
+from multiversx_sdk.network_providers.transactions import TransactionOnNetwork
+
+
+class IQuery(Protocol):
+    def get_contract(self) -> IAddress:
+        ...
+
+    def get_function(self) -> str:
+        ...
+
+    def get_encoded_arguments(self) -> Sequence[str]:
+        ...
+
+    def get_caller(self) -> Optional[IAddress]:
+        ...
+
+    def get_value(self) -> int:
+        ...
+
+
+class IQueryResponse(Protocol):
+    return_data: List[str]
+    return_code: str
+    return_message: str
+    gas_used: int
+
+    def get_return_data_parts(self) -> List[bytes]:
+        ...
+
+
+class INetworkConfig(Protocol):
+    chain_id: str
+
+
+class INetworkProvider(Protocol):
+    def get_network_config(self) -> INetworkConfig:
+        ...
+
+    def query_contract(self, query: IQuery) -> IQueryResponse:
```

Comment:
In `feat/next`, we should drop the old types on queries (cleanup), and only use the new ones:

https://github.com/multiversx/mx-sdk-py/blob/main/multiversx_sdk/core/smart_contract_query.py

If possible, can adjust the network provider in this PR, otherwise, should be done in a future PR.

Reply:
Will do it in a future PR.